pkg/runner/runner.go

```diff
@@ -154,10 +157,46 @@ func RunBundle(action string, ns string, bundleName string, sandboxRole string,
 		return
 	}
 	fmt.Printf("Successfully created pod [%v] to %s [%v] in namespace [%v]\n", pn, ec.Action, bundleName, ns)
+
+	if printLogs {
+		printBundleLogs(pn, ns, action)
+	}
+
 	// TODO: return nil
 	return
 }
 
+func printBundleLogs(podName string, namespace string, action string) {
+	k8scli, err := clients.Kubernetes()
+	if err != nil {
+		panic(err.Error())
+	}
+
+	logTailRequest := k8scli.Client.CoreV1().Pods(namespace).GetLogs(podName, &v1.PodLogOptions{Follow: true})
+
+	var requestStream io.ReadCloser
+	var podStarted bool
+
+	for podStarted == false {
+		requestStream, err = logTailRequest.Stream()
+		if err != nil {
+			fmt.Printf("Waiting for bundle [%v] pod [%v] to start...\n", action, podName)
+			log.Debugf("%v", err)
+			time.Sleep(3 * time.Second)
+		} else {
+			fmt.Printf("Pod started. Reading logs...\n")
+			podStarted = true
+		}
+	}
+	defer requestStream.Close()
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(requestStream)
+	fmt.Println("-+- --------------------- -+-")
+	fmt.Println(" |       BUNDLE LOGS       | ")
+	fmt.Println("-+- --------------------- -+-")
+	fmt.Println(buf.String())
```

Comment:
So does this just read the whole thing before printing? Feels like if we are actively reading from a stream we should be writing the stream as it comes in. Right now we're reading it all into a string then dumping it. So I can't really "watch it". 

Basically, we want to spawn a goroutine that reads from the stream and writes the output to a channel. Then have the main routine block on that channel reading any data that was written to it and write it to stdout. Once the stream is done, the channel should be signaled to close, then the main routine can proceed.

Reply:
That is correct, this reads the whole thing. I was having trouble figuring out how to print logs as they are available. I'll ping you on IRC.